dolphin/api/v1/storages.py

```diff
@@ -132,8 +132,21 @@ def create(self, req, body):
     def update(self, req, id, body):
         return dict(name="Storage 4")
 
+    @wsgi.response(202)
     def delete(self, req, id):
-        return webob.Response(status_int=http_client.ACCEPTED)
+        ctxt = req.environ['dolphin.context']
+        try:
+            device = db.storage_get(ctxt, id)
+        except Exception as e:
+            LOG.error(e)
+            raise exc.HTTPBadRequest(explanation=e.msg)
+        else:
+            for subclass in task.StorageResourceTask.__subclasses__():
+                self.task_rpcapi.remove_storage_resource(
+                    ctxt,
+                    id,
```

Comment:
instead of id, how about storage['id']??

Reply:
Where can we get the object 'storage'

Reply:
sorry, i mean device['id']

Reply:
I think maybe directly use id is better, because id is one of the parameter of 'delete(self, req, id)', so use this 'id' to call other functions in 'delete' function maybe more intuitive.
Maybe change 'delete(self, req, **id**)' -> 'delete(self, req, **storage_id**)' is another option.

Reply:
Done